src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/NuGetPowerShellUsageCollector.cs

```diff
@@ -0,0 +1,371 @@
+            internal int PMCWindowLoadCount { get; set; }
+            internal int PMUIExecuteCommandCount { get; set; }
+            internal bool SolutionLoaded { get; set; }
+
+            internal SolutionData()
+            {
+                FirstTimeLoadedFromPMC = false;
+                FirstTimeLoadedFromPMUI = false;
+                InitPs1LoadedFromPMCFirst = false;
+                InitPs1LoadPMC = false;
+                InitPs1LoadPMUI = false;
+                LoadedFromPMC = false;
+                LoadedFromPMUI = false;
+                NuGetCommandUsed = false;
+                PMCExecuteCommandCount = 0;
+                PMCWindowLoadCount = 0;
+                PMUIExecuteCommandCount = 0;
+                SolutionLoaded = false;
+
+            }
+
+            internal TelemetryEvent ToTelemetryEvent()
+            {
+                var telemetry = new TelemetryEvent(NuGetVSSolutionClose,
+                    new Dictionary<string, object>()
+                    {
+                        { NuGetPowershellPrefix + NuGetPowerShellUsageCollector.FirstTimeLoadedFromPMC , FirstTimeLoadedFromPMC },
+                        { NuGetPowershellPrefix + NuGetPowerShellUsageCollector.FirstTimeLoadedFromPMUI , FirstTimeLoadedFromPMUI },
+                        { NuGetPowershellPrefix + NuGetPowerShellUsageCollector.InitPs1LoadedFromPMCFirst , InitPs1LoadedFromPMCFirst },
+                        { NuGetPowershellPrefix + NuGetPowerShellUsageCollector.InitPs1LoadPMC , InitPs1LoadPMC },
+                        { NuGetPowershellPrefix + NuGetPowerShellUsageCollector.InitPs1LoadPMUI , InitPs1LoadPMUI },
+                        { NuGetPowershellPrefix + NuGetPowerShellUsageCollector.LoadedFromPMC , LoadedFromPMC },
+                        { NuGetPowershellPrefix + NuGetPowerShellUsageCollector.LoadedFromPMUI , LoadedFromPMUI },
+                        { NuGetPowershellPrefix + NuGetPowerShellUsageCollector.NuGetCommandUsed , NuGetCommandUsed },
+                        { NuGetPowershellPrefix + NuGetPMCExecuteCommandCount , PMCExecuteCommandCount },
+                        { NuGetPowershellPrefix + NuGetPMCWindowLoadCount , PMCWindowLoadCount },
+                        { NuGetPowershellPrefix + NuGetPMUIExecuteCommandCount , PMUIExecuteCommandCount },
+                        { NuGetPowershellPrefix + NuGetPowerShellUsageCollector.SolutionLoaded , SolutionLoaded }
+                    });
+
+                return telemetry;
+            }
+        }
+
+        internal class InstanceData
+        {
+            internal int PMCExecuteCommandCount { get; set; }
+            internal int PMCWindowLoadCount { get; set; }
+            internal int PMUIExecuteCommandCount { get; set; }
+            internal int PMCLoadedSolutionCount { get; set; }
+            internal int PMUILoadedSolutionCount { get; set; }
+            internal bool ReOpenAtStart { get; set; }
+            internal int SolutionCount { get; set; }
```

Comment:
Why are there 3 solution counts?  I don't understand how any of them could have different values.